platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -365,7 +365,15 @@ public UsersWorkspace getWorkspace(String name, String owner) throws BadRequestE
 
     /*******************************/
 
-    void startWorkspaceAsync(final UsersWorkspaceImpl usersWorkspace, final String envName) {
+    void startWorkspaceAsync(final UsersWorkspaceImpl usersWorkspace, final String envName) throws ConflictException {
+        try {
```

Comment:
If you move this check in the methods that are call startWorkspaceAsync you will prevent calling `workspaceDao.get(workspaceName, owner)`. WDYT?